cpp/src/arrow/json/parser.cc

```diff
@@ -454,6 +462,8 @@ class RawBuilderSet {
         }
         return Status::OK();
       }
+      case Kind::kNumberOrString:
+        return MakeBuilder<Kind::kNumberOrString>(leading_nulls, builder);
```

Comment:
For clarity, move this up just below the `kString` case?

Reply:
done